snuba/utils/metrics/backends/datadog.py

```diff
@@ -12,16 +13,30 @@ class DatadogMetricsBackend(MetricsBackend):
     """
 
     def __init__(
-        self, client: DogStatsd, sample_rates: Optional[Mapping[str, float]] = None
+        self,
+        client_factory: Callable[[], DogStatsd],
+        sample_rates: Optional[Mapping[str, float]] = None,
     ) -> None:
         """
+        :param client_factory: A function that returns a new ``DogStatsd``
+        instance. (These instances are not thread safe, so a new instance
+        will be created for each independent thread.)
```

Comment:
Aside: it might be nice to start generating API documentation?

Reply:
I am fully on board with anything that produces reasonable documentation without us having to write it twice.